src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvoker.cs

```diff
@@ -346,7 +346,7 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
                     arguments,
                     controller);
                 logger.ActionMethodExecuting(controllerContext, orderedArguments);
-
+                var stopWatch = Stopwatch.StartNew();
```

Comment:
Take a look at `ValueStopWatch` https://github.com/aspnet/HttpClientFactory/blob/dev/src/Microsoft.Extensions.Http/Logging/LoggingHttpMessageHandler.cs#L36

Reply:
ValueStopwatch is an internal struct so I can't use that in Mvc right? Did you mean the stopwatch I use should have similar/same code as ValueStopwatch?

Reply:
FYI, Resource invoker using StopWatch here: https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.Core/Internal/ResourceInvoker.cs#L72, probably we should be consistent across

Reply:
Using the stopwatch like ResourceInvoker does to keep it consistent.

Reply:
ValueStopWatch is shared source https://github.com/aspnet/Common/tree/dev/shared/Microsoft.Extensions.ValueStopwatch.Sources